db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/OptimizerImpl.java

```diff
@@ -2566,6 +2560,55 @@ private void costBasedCostOptimizable(Optimizable optimizable,
         }
     }
 
+    private static boolean isCurrentAccessPathBetter(AccessPath currentAccessPath, AccessPath bestAccessPath,
+                                                     Optimizable currOpt, OptimizablePredicateList predList) throws StandardException {
+        CostEstimate currentCost = currentAccessPath.getCostEstimate();
+        CostEstimate bestCost = bestAccessPath.getCostEstimate();
+
+        if ((bestCost == null) || bestCost.isUninitialized()) {
+            return true;
+        }
+
+        double compResult = currentCost.compare(bestCost);
+        if (compResult < 0) {
+            return true;
+        } else if (compResult == 0) {
```

Comment:
But which other cases do you have in mind that we should also fix? I don't quite get what does it mean by having no estimates. I mean, if code reaches here, both `currentCost` and `bestCost`should be valid estimates.

In my opinion, this is a special case anyway because statistics is collected when the table is empty. Otherwise I think the chance that two access paths' costs are the same is very low. I mean, even it happens in that case, the two access paths should have similar performance. Otherwise it's a problem of the cost model.

Having the uncertainty factor would bring at least the following questions:
- What's a good uncertainty factor?
- With the uncertainty factor, we generally over estimates on row count. What about the boundary queries that having the uncertainty factor lead to a slower plan because of higher row count estimate?
- How do we print this info in explain?